framework/Processors/Factory.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Created on July 10, 2013
+
+@author: alfoa
+"""
+
+#for future compatibility with Python 3-----------------------------------------
+from __future__ import division, print_function, unicode_literals, absolute_import
+import warnings
+warnings.simplefilter('default',DeprecationWarning)
+if not 'xrange' in dir(__builtins__):
+  xrange = range
+#End compatibility block for Python 3-------------------------------------------
+
+################################################################################
+from .Processor import Processor
+from .BasicStatistics import BasicStatistics
+from .ComparisonStatistics import ComparisonStatistics
+from .ExternalProcessor import ExternalProcessor
+from .ImportanceRank import ImportanceRank
+from .InterfacedProcessor import InterfacedProcessor
+from .LimitSurface import LimitSurface
+from .LimitSurfaceIntegral import LimitSurfaceIntegral
+from .RavenOutput import RavenOutput
+from .SafestPoint import SafestPoint
+
+from .TopologicalDecomposition import TopologicalDecomposition
+from .DataMining import DataMining
+
+try:
+  from .TopologicalDecomposition import QTopologicalDecomposition
+  from .DataMining import QDataMining
+except ImportError:
+  pass
```

Comment:
Any chance we could add a brief comment here explaining why we're wrapping these imports in a `try`?